Automattic/wp-calypso
# Simplify Jetpack's signup flow with magic links

#### Changes proposed in this Pull Request

* Replace the Jetpack's sign-up form with an automatic account creation based on magic links. The idea is to decrease the friction users have to go through in order to setup Jetpack. Instead of asking users to pick an email address, create a username and a password, we will only prompt them for an email address and create the account behind the scene.

#### Notes

* Do not focus on the copy. This is proof of concept focused only on providing a quick version of new the signup flow.

#### Testing instructions

* Download this PR.
* Run `yarn start` to fire up Calypso Blue.
* Open an incognito window (Google Chrome or Firefox are good browser choices).
* Create a Jurassic Ninja site or get a self-hosted site.
* Initiate Jetpack's setup flow.
* Make sure that you're redirected to `/log-in/jetpack` (we don't want to go through the in-place connection flow).
* Copy your browser URL and paste it in a new tab (we are going to test different flows so keep a copy of this first URL).

**Login flow (or nothing has changed)**
* Enter a valid username or email address that has a corresponding WPCOM account.
* Complete the login flow.
* Make sure that you're logged in and everything worked exactly as in production.

**Signup flow (email provided)**
* Enter a valid email address that **doesn't** have a corresponding WPCOM account.
* Verify that you're redirected to the magic link confirmation page (it tells you that an email was sent to you).
* Go to your email inbox and click the CTA of the email you received.
* Verify that a WPCOM account was created and linked to the email you provided and that you're logged in to the application.

**Signup flow (username provided)**
* Enter a username that **doesn't** have a corresponding WPCOM account.
* Verify that you're redirected to the magic link form page (you're prompt for an email address).
* Enter a valid email address that doesn't have a corresponding WPCOM account.
* Verify that you're redirected to the magic link confirmation page (it tells you that an email was sent to you).
* Go to your email inbox and click the CTA of the email you received.
* Verify that a WPCOM account was created and linked to the email you provided and that you're logged in to the application.

Related to 1199916399796129-as-1199945743087583

#### Demo
https://user-images.githubusercontent.com/3418513/109978193-625f6380-7cdc-11eb-8cff-c1b6260c8c0a.mp4

#### Screenshots for translators
**Log in or create a WordPress.com account to get started with Jetpack**
<img width="462" alt="T1" src="https://user-images.githubusercontent.com/3418513/110352929-41b54780-8015-11eb-846e-cede2da55eac.png">

**If you continue with Google, Apple, or an email that isn’t registered yet, you are creating a new WordPress.com account.**
<img width="462" alt="T2" src="https://user-images.githubusercontent.com/3418513/110352922-3feb8400-8015-11eb-992e-8a05fbff2c2e.png">

**We just emailed a link to {{strong}}%(emailAddress)s{{/strong}}.**
<img src="https://user-images.githubusercontent.com/3418513/110352906-3bbf6680-8015-11eb-8d88-ee4a8a516dc0.png" width="462" />
